codegen/src/main.rs

```diff
@@ -0,0 +1,268 @@
+                            let post_body = serde_json::to_string(&request).unwrap();
+                            let stamp = stamp(post_body.clone(), &self.api_key).unwrap();
+
+                            let res = self.http
+                                .post(url)
+                                .header("X-Stamp", stamp)
+                                .body(post_body)
+                                .send()
+                                .await?;
+                            let parsed = res.json::<coordinator::{res_type}>().await?;
+                            Ok(parsed)
+                        }}
+                    "#
+                    );
+                    println!("Generating {} (query)", fn_name);
+                    generated_methods.push_str(&func);
+                }
+            }
+        }
+    }
+
+    // Insert the generated methods into the right spot inside of our template
+    let method_placeholder = "// <GENERATED_METHODS_PLACEHOLDER>";
+    let client_template = fs::read_to_string(CLIENT_TEMPLATE_PATH).unwrap();
+    let output = client_template.replace(method_placeholder, &generated_methods);
+
+    // Write the generated client to the `client` crate
+    let generated_client_path = PathBuf::from(GENERATED_CLIENT_DIR).join("client.rs");
+    fs::create_dir_all("../client/src/generated").unwrap();
+    fs::write(generated_client_path, output).expect("Failed to write generated file");
+
+    // Now add client.rs to the generated/mod.rs file
+    let mut mod_rs = fs::OpenOptions::new()
+        .append(true)
+        .open(out_dir.join("mod.rs"))
+        .unwrap();
+
+    // This will append a new line to the end of the file
+    writeln!(mod_rs, "\n// Added by tkhq_codegen").unwrap();
+    writeln!(mod_rs, "mod client;").unwrap();
+    writeln!(mod_rs, "pub use services::coordinator::public::v1::*;").unwrap();
+    writeln!(mod_rs, "pub use external::activity::v1::*;").unwrap();
+    writeln!(mod_rs, "pub use immutable::activity::v1::*;").unwrap();
+
+    // Now, are you ready for this? tonic/prost do not support `enums` in ergonomic forms.
+    // A field that is an enum is represented as an i32 :(
+    // Prost has annotations for this, but serde has no clue and can't do anything about this:
+    //   #[prost(enumeration = "super::super::common::v1::ApiKeyCurve", tag = "3")]
+    //   pub curve_type: i32,
+    // ...and prost doesn't support JSON serialization...
+    // So here we are, doing some post-processing on the generated code, to manually switch enum fields from
+    // the generic "i32" type to their _actual_ type (e.g. super::super::common::v1::ApiKeyCurve)
+    // files that we need to add custom JSON logic to
+    for entry in WalkDir::new(GENERATED_CLIENT_DIR)
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| {
+            e.path().extension().map_or(false, |ext| ext == "rs")
+                && !e.file_name().to_string_lossy().contains("google.api")
```

Comment:
There are two vendored google files: `google.api.rs` and `google.rpc.rs`, is it intentional that this statement is only excluding `google.api.rs`

Reply:
<img width="359" alt="image" src="https://github.com/user-attachments/assets/3dd4fb71-9929-4da9-a5b5-b04fc32d88c4" />

Reply:
Ah yes. I'm not sure why I kept this processed. I don't need it processed! Fixed.